backend/src/test/java/com/backend/backend/BookingServiceTest.java

```diff
@@ -0,0 +1,153 @@
+        );
+
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @Test
+    public void addBooking_Happy() {
+        // Arrange
+
+        Booking booking = new Booking();
+        booking.setPid("P12312");
+        booking.setUid("U456");
+        booking.setBookingDate(new Date());
+        booking.setStatus("Confirmed");
+        booking.setNo_of_persons(2);
+        ;
+
+        Mockito.when(normalUser.getBookingHistory()).thenReturn(new ArrayList<>());
+        Mockito.when(normalUserRepository.findById(Mockito.any(String.class))).thenReturn(Optional.of(normalUser));
+        Mockito.when(normalUserRepository.save(normalUser)).thenReturn(normalUser);
+        Mockito.when(destinationRepository.findById(Mockito.any(String.class))).thenReturn(Optional.of(new Destination()));
+
+        // Act
+        String result = bookingService.addBooking(booking);
+
+        // Assert
+        assertEquals("Booking added successfully!", result);
+    }
+
+
+    @Test
+    public void addBooking_sad_1() {
+        // Arrange
+
+        Booking booking = new Booking();
+        booking.setPid("P12312");
+        booking.setUid("U456");
+        booking.setBookingDate(new Date());
+        booking.setStatus("Confirmed");
+        booking.setNo_of_persons(2);
+
+        Mockito.when(normalUserRepository.findById(Mockito.any(String.class))).thenThrow(new NoSuchElementException("User not found!"));
+
+        // Act
+        String result = bookingService.addBooking(booking);
+
+        // Assert
+        assertEquals("Error: User not found!", result);
+    }
+
+    @Test
+    public void addBooking_sad_2() {
+        // Arrange
+
+        Booking booking = new Booking();
+        booking.setPid("P12312");
+        booking.setUid("U456");
+        booking.setBookingDate(new Date());
+        booking.setStatus("Confirmed");
+        booking.setNo_of_persons(2);
+
+        Mockito.when(normalUserRepository.findById(Mockito.any(String.class))).thenReturn(Optional.of(normalUser));
+        Mockito.when(destinationRepository.findById(Mockito.any(String.class))).thenThrow(new NoSuchElementException("Destination not found!"));
+
+
+        // Act
+        String result = bookingService.addBooking(booking);
+
+        // Assert
+        assertEquals("Error: Destination not found!", result);
+    }
+
+    @Test
+    public void addBooking_sad_3() {
+        // Arrange
+
+        Booking booking = new Booking();
+        booking.setPid("P12312");
+        booking.setUid("U456");
+        booking.setBookingDate(new Date());
+        booking.setStatus("Confirmed");
+        booking.setNo_of_persons(2);
+
+        Mockito.when(normalUserRepository.findById(Mockito.any(String.class))).thenThrow(new DuplicateFormatFlagsException(""));
+
+
+        // Act
+        String result = bookingService.addBooking(booking);
+
+        // Assert
+        assertEquals("An unexpected error occurred!", result);
+    }
+
+}
```

Comment:
The `BookingServiceTest` class thoroughly tests the `addBooking` method with both happy and sad paths, including scenarios for missing user, missing destination, and unexpected errors. However, the tests could benefit from improved maintainability by refactoring common setup steps into reusable helper methods or using parameterized tests for error scenarios to reduce redundancy.